lib/modules/manager/tekton/extract.ts

```diff
@@ -0,0 +1,97 @@
+import is from '@sindresorhus/is';
+import { loadAll } from 'js-yaml';
+import { logger } from '../../../logger';
+import { getDep } from '../dockerfile/extract';
+import type { PackageDependency, PackageFile } from '../types';
+import type { TektonBundle, TektonResource } from './types';
+
+export function extractPackageFile(
+  content: string,
+  fileName: string
+): PackageFile | null {
+  logger.trace('tekton.extractPackageFile()');
+  const deps: PackageDependency[] = [];
+  let docs: any[];
+  try {
+    docs = loadAll(content) as TektonResource[];
+  } catch (err) {
+    logger.debug(
+      { err, fileName },
+      'Failed to parse YAML resource as a Tekton resource'
+    );
+    return null;
+  }
+  for (const doc of docs) {
+    deps.push(...getDeps(doc));
+  }
+  if (!deps.length) {
+    return null;
+  }
+  return { deps };
+}
+
+function getDeps(doc: TektonResource): PackageDependency[] {
+  const deps: PackageDependency[] = [];
+
+  // Handle TaskRun resource
+  addDep(doc?.spec?.taskRef, deps);
```

Comment:
doc should never undefined or null here, so please all following
```suggestion
  addDep(doc.spec?.taskRef, deps);
```